test/per_file_coverage.sh

```diff
@@ -70,6 +70,8 @@ declare -a KNOWN_LOW_COVERAGE=(
 "source/extensions/io_socket:96.2"
 "source/extensions/io_socket/user_space:96.2"
 "source/extensions/network/dns_resolver/getaddrinfo:96.3"
+"source/extensions/path/pattern_template_lib:93.2"
```

Comment:
Normally we should be able to get ~100% code coverage from unit tests. Do you know which lines are not being covered?

Reply:
There are failure cases that are very difficult to cover, such as regex engine failure. This coverage is the same as internal google + a few extra I added as well.

Reply:
If you look at the coverage report here:
https://storage.googleapis.com/envoy-pr/c8b8f01/coverage/source/extensions/path/pattern_template_lib/pattern_template.cc.gcov.html
and here:
https://storage.googleapis.com/envoy-pr/c8b8f01/coverage/source/extensions/path/pattern_template_lib/pattern_template_internal.cc.gcov.html
It looks like increasing the unit test coverage to get to 100% should be quite easy. For example, in pattern_template.cc there are no tests for:

isValidMatchPattern
isValidPathTemplateRewritePattern
isValidSharedVariableSet

And there is some missing coverage for some of the error cases. Please add unit tests for all methods.